cmd/frontend/graphqlbackend/search_results_test.go

```diff
@@ -1308,3 +1309,74 @@ func TestEvaluateAnd(t *testing.T) {
 		})
 	}
 }
+
+func TestSearchContext(t *testing.T) {
+	orig := envvar.SourcegraphDotComMode()
+	envvar.MockSourcegraphDotComMode(true)
+	defer envvar.MockSourcegraphDotComMode(orig)
+
+	tts := []struct {
+		name        string
+		searchQuery string
+		numContexts int
+	}{
+		{name: "single search context", searchQuery: "foo context:@userA", numContexts: 1},
+		{name: "multiple search contexts", searchQuery: "foo (context:@userA or context:@userB)", numContexts: 2},
+	}
+
+	users := map[string]int32{
+		"userA": 1,
+		"userB": 2,
+	}
+
+	mockZoekt := &searchbackend.Zoekt{
+		Client:       &searchbackend.FakeSearcher{Repos: []*zoekt.RepoListEntry{}},
+		DisableCache: true,
+	}
+
+	for _, tt := range tts {
+		t.Run(tt.name, func(t *testing.T) {
+			qinfo, err := query.ProcessAndOr(tt.searchQuery, query.ParserOptions{})
```

Comment:
```suggestion
			qinfo, err := query.ParseLiteral(tt.searchQuery)
```

`ParseLiteral` is just a helper parser for search queries where we know it's just literals tested, it'll still work with or-operators.